### PR TITLE
Replace pointers-to-pointer with pointers.

### PR DESCRIPTION
**Pull request by d**
We've got quite a few places that curiously use a pointer-to-a-pointer where a simple pointer would suffice. This antipattern confuses humans and programs alike: "star-star" parameters are a strong signal for output parameters, but we don't do any "outputting" in some cases. This patch replaces all of them with pointers when applicable.

### CODE REVIEW

File: src/backend/gporca/libgpopt/include/gpopt/xforms/CXformUtils.h [hunk 1]
```diff
@@ -87,9 +87,6 @@ class CXformUtils
 											   CTableDescriptor *ptabdesc,
 											   COperator *popLogical);
 
-	// private copy ctor
-	CXformUtils(const CXformUtils &);
```
Comment (hardikar) [inline src/backend/gporca/libgpopt/include/gpopt/xforms/CXformUtils.h:91]: Do you want to delete this instead?

**Review by hardikar — APPROVED** [review]

**Review by zellerh — APPROVED** [review]
LGTM, thanks for deleting some dead code as well.

**Review by l-wang — APPROVED** [review]
LGTM. Thanks for cleaning up the code!

**Comment by d** [comment]
Thanks for everyone's time for the review, pushed

**Pull request merged**